### PR TITLE
Address some reviewer points for SymPy article

### PR DESCRIPTION
**Pull request by certik**
Posting on behalf of @skirpichev:

Below some todo lists for review comments.

reviewer 1
- [x] 1) A part that confused me is the assertion (footnote 3)...
  - This solved, at least partially (I don't sure if we
    should explain in details how to define commuatators etc).
- [ ] 2) References reasonable, though [8], an excellent reference ...
  -  Not sure how to solve this yet.  Perhaps, this is a
    bad reference?

reviewer 2
- [x] 1) >>> circuit_plot(fourier, nqubits=3) ...
  -   Notebook reformated (as nbformat 3)
- [x] 2) Line 105...
  -  Done.
- [x] 3) Lines 151-153
  -  Added a footnote about dotprint
- [ ] 4) Section 2.3 - Assumptions...
  -  Not sure.  Looks like a question about documentation?
    Or foo.assumptions0?  Or foo.is_<tab>, i.e. how
    user can discover assumptions on the object?
- [x] 5) Section 2.4...
  -  Suggestion used.
- [x] 6) Line 221:...
  -  Done.  Footnote moved, perhaps this is better.
- [x] 7) Page 8: Footnote 5
  -  Suggestion used.
- [x] 8) Section 4.1... Should mpmath be cited here?
  -  Citation added.

reviewer 3

Nothing yet.  But I feel, that we can address some points
on pp2-6 before major rewrite, suggested on p1.
What do you think?

### CODE REVIEW

**Comment by certik** [comment]
Thanks @skirpichev for fixing lots of the issues, much appreciated. Yes, I was thinking that too to first fix all the comments except the major rewrite, so that we satisfy all the feedback that we got. And then concentrate on the major rewrite/reorganization feedback, and see what to do about it.

**Comment by asmeurer** [comment]
For the all assumptions question, the best reference is http://docs.sympy.org/latest/modules/core.html#sympy.core.assumptions

File: architecture.tex [hunk 1]
```diff
@@ -10,7 +10,12 @@ \subsection{Basic Usage}
 % symbols, various ways to declare them
 
 The following statement
-imports all SymPy functions into the global Python namespace.
+imports all SymPy functions into the global Python namespace.\footnote{
+Nevertheless, it is best to avoid such wildcard import statements
```
Comment (asmeurer) [inline architecture.tex:14]: Maybe some note like "This is used here to improve readability of the examples."

File: architecture.tex [hunk 2]
```diff
@@ -102,7 +102,10 @@ \subsection{The Core}
 \end{verbatim}
 
 A useful way to view an expression tree is using the \texttt{srepr} function, which
-returns a string representation of an expression as valid Python code
+returns a string representation of an expression as valid Python code\footnote{
+The \texttt{dotprint} function from the \texttt{sympy.printing.dot} package
+prints output to dot format, which can be rendered with Graphviz to
```
Comment (asmeurer) [inline architecture.tex:107]: We may need to cite Graphviz.

**Comment by asmeurer** [comment]
Looks good. Let's move the checklist to an issue so we can track all the reviewer comments.

**Comment by certik** [comment]
Thanks @asmeurer for reviewing and merging.

**Pull request merged**